gunpowder/jax/nodes/train.py

```diff
@@ -0,0 +1,288 @@
+import logging
+import numpy as np
+import jax
+import jax.numpy as jnp
+import pickle
+import os
+
+from gunpowder.array import ArrayKey, Array
+from gunpowder.array_spec import ArraySpec
+from gunpowder.ext import tensorboardX, NoSuchModule
+from gunpowder.nodes.generic_train import GenericTrain
+from gunpowder.jax import GenericJaxModel
+
+from typing import Dict, Union, Optional
+
+
+logger = logging.getLogger(__name__)
+
+
+class Train(GenericTrain):
+    """JAX implementation of :class:`gunpowder.nodes.GenericTrain`.
+
+    Args:
+
+        model (subclass of ``gunpowder.jax.GenericJaxModel``):
+
+            The model to train. This model encapsulates the forward model,
+            loss, and optimizer.
+
+        inputs (``dict``, ``string`` -> :class:`ArrayKey`):
+
+            Dictionary from the names of input tensors (argument names of the
```

Comment:
Docstring doesn't seem to match the supported values of inputs that `__collect_provided_arrays` implementation seems to imply. would `Any` -> `Union[str, np.ndarray, ArrayKey]` be more accurate?

Reply:
Yup that should be more accurate

Reply:
Thinking about it a little, I think support should only for mapping ``string`` to `ArrayKey` or np.ndarray. Other usages might be possible but unsupported for now. Same for output.

Reply:
Sounds good to me